<launch>

    <param name="uav_model" command="$(find xacro)/xacro $(find uav_swarm_description)/urdf/quadrotor.urdf.xacro" />

    <arg name="pluginlists_yaml" default="$(find mavros)/launch/apm_pluginlists.yaml" />
    <arg name="config_yaml" default="$(find uav_swarm_control)/launch/apm_config.yaml" />


    <arg name="uav0_id" default="0" />
    <arg name="uav1_id" default="1" />
    <arg name="uav2_id" default="2" />

    <arg name="origin_lat" default="-22.413949" />
    <arg name="origin_lon" default="-45.446555" />

    <param name="/uav_swarm_control/origin_lat" value="$(arg origin_lat)" />
    <param name="/uav_swarm_control/origin_lon" value="$(arg origin_lon)" />
    <param name="/uav_swarm_control/fix_topic" value="mavros/global_position/global"/>
    <param name="/uav_swarm_control/odom_topic" value="mavros/local_position/odom"/>
    <param name="/uav_swarm_control/cmd_vel_topic" value="mavros/setpoint_velocity/cmd_vel"/>

    <group ns="uav0">
        <param name="uav_id" value="$(arg uav0_id)" />
        <param name="tf_frame" value="uav$(arg uav0_id)"/>

        <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" >
                <param name="fcu_url" value="udp://127.0.0.1:14551@" />
                <param name="gcs_url" value="" />
                <param name="target_system_id" value="1" />
                <param name="target_component_id" value="1" />

                <!-- load blacklist, config -->
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(arg config_yaml)" />
        </node>

        <node pkg="uav_swarm_control" type="swarm_controller_node" name="swarm_controller_node" output="screen" />

        <node pkg="uav_swarm_control" type="marker_publisher" name="marker_publisher" />
    </group>

    <group ns="uav1">
            <param name="uav_id" value="$(arg uav1_id)" />
            <param name="tf_frame" value="uav$(arg uav1_id)"/>

            <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" >
                    <param name="fcu_url" value="udp://127.0.0.1:14561@" />
                    <param name="gcs_url" value="" />
                    <param name="target_system_id" value="1" />
                    <param name="target_component_id" value="1" />

                    <!-- load blacklist, config -->
                    <rosparam command="load" file="$(arg pluginlists_yaml)" />
                    <rosparam command="load" file="$(arg config_yaml)" />
            </node>

            <node pkg="uav_swarm_control" type="swarm_controller_node" name="swarm_controller_node" output="screen" />

            <node pkg="uav_swarm_control" type="marker_publisher" name="marker_publisher" />
    </group>

    <group ns="uav2">
        <param name="uav_id" value="$(arg uav2_id)" />
        <param name="tf_frame" value="uav$(arg uav2_id)"/>

        <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" >
                <param name="fcu_url" value="udp://127.0.0.1:14571@" />
                <param name="gcs_url" value="" />
                <param name="target_system_id" value="1" />
                <param name="target_component_id" value="1" />

                <!-- load blacklist, config -->
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(arg config_yaml)" />
        </node>

        <node pkg="uav_swarm_control" type="swarm_controller_node" name="swarm_controller_node" output="screen" />

        <node pkg="uav_swarm_control" type="marker_publisher" name="marker_publisher" />
    </group>

    <!-- Run RVIZ-->
    <node name="$(anon rviz)" pkg="rviz" type="rviz" args="-d $(find uav_swarm_control)/uav_swarm_control.rviz" />

</launch>
